exo/topology/ring_memory_weighted_partitioning_strategy.py

```diff
@@ -9,10 +9,14 @@ def partition(self, topology: Topology) -> List[Partition]:
     nodes = list(topology.all_nodes())
     nodes.sort(key=lambda x: (x[1].memory, x[0]), reverse=True)
     total_memory = sum(node[1].memory for node in nodes)
+    total_flops = sum(node[1].flops.fp32 for node in nodes)
     partitions = []
     start = 0
     for node in nodes:
-      end = round(start + (node[1].memory/total_memory), 5)
+      memory_weight = node[1].memory / total_memory
+      flops_weight = node[1].flops.fp32 / total_flops
+      weight = (memory_weight + flops_weight) / 2
+      end = round(start + weight, 5)
```

Comment:
this breaks the partitioning strategy interface. you need to return a contiguous set of ranges from 0 to 1